### PR TITLE
How to

### PR DESCRIPTION
**Pull request by willmcgugan**
Adds a How To section, and a "How to design a layout" article.

Open to suggestions for new How-Tos.

### CODE REVIEW

File: docs/how-to/design-a-layout.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+When you dock a widget, it reduces the available area for other widgets.
+This means that Textual will automatically compensate for the 6 additional lines reserved for the header and footer.
+
+## Tip 4. Use FR Units for flexible things
+
+After we've added the header and footer, we want the remaining space to be used for the main interface, which will contain the columns in the sketch.
+This area is flexible (will change according to the size of the terminal), so how do we ensure that it takes up precisely the space needed?
+
+The simplest way is to use [fr](../css_types/scalar.md) units.
```
Comment (rodrigogiraoserrao) [inline docs/how-to/design-a-layout.md:93]:
```suggestion
The simplest way is to use [fr](../css_types/scalar.md#fraction) units.
```

File: docs/how-to/design-a-layout.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    ```python hl_lines="25-32 36-46"
+    --8<-- "docs/examples/how-to/layout06.py"
+    ```
+
+=== "Output"
+
+    ```{.textual path="docs/examples/how-to/layout06.py" columns="100" lines="32"}
+    ```
+
+You should see from the output that we have fixed with columns, that will scroll horizontally.
```
Comment (davep) [inline docs/how-to/design-a-layout.md:177]:
```suggestion
You should see from the output that we have fixed width columns, that will scroll horizontally.
```

File: docs/how-to/index.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# How To
+
+Welcome to the How To section.
+
+Here you will find How To articles which cover various topics at a higher level than the Guide or Reference.
```
Comment (davep) [inline docs/how-to/index.md:5]: Perhaps link to the guide and reference here?

File: docs/how-to/design-a-layout.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    ```{.textual path="docs/examples/how-to/layout04.py"}
+    ```
+
+The container will appear as blank space until we add some widgets to it.
+
+Let's add the columns to the `HorizontalScroll`.
+A column is itself a container which will can have a vertical scrollbar, so we will define our `Column` by subclassing `VerticalScroll`.
+In a real app, these columns will likely be added dynamically from some kind of configuration, but let's add 4 to visualize the layout.
+
+We will also define a `Tweet` placeholder and add a few to each column,
```
Comment (rodrigogiraoserrao) [inline docs/how-to/design-a-layout.md:143]:
```suggestion
We will also define a `Tweet` placeholder and add a few to each column.
```

File: docs/how-to/design-a-layout.md (hunk 2 above)
Comment (rodrigogiraoserrao) [inline docs/how-to/design-a-layout.md:177]:
```suggestion
You should see from the output that we have fixed width columns that will scroll horizontally.
```

**Review by rodrigogiraoserrao — APPROVED** [review]
Nice! Working outside in feels like a good tip.

File: docs/how-to/index.md (hunk 3 above)
Comment (willmcgugan) [reply docs/how-to/index.md:5]: The reader can bloody well just look up at the menu bar.

Comment (rodrigogiraoserrao) [reply docs/how-to/index.md:5]: 😆

**Pull request merged**